### DISCUSSION TITLE
Arduino UNO R4 Minima

### DISCUSSION THREAD

**Discussion by tpiuser** [Sloeber/arduino-eclipse-plugin#1620] [General]
Does anyone know if Sloeber 4.4.2 (or before) supports the Arduino UNO R4 Minima board?  
If so, how do I go about selecting that board?

**Comment by jantje** [comment]
It should be
Just like in Arduino ide you will have to install the boards in the boards manager
![image](https://github.com/Sloeber/arduino-eclipse-plugin/assets/1175841/50b1a5c9-e31d-4a71-aa52-f1f4ca676b99)
If you do not know how to do so please see this documentation https://github.com/Sloeber/arduino-eclipse-plugin/discussions/1555

Once you installed the boards you should be able to select them in the new sketch wizard and in the project properties
![image](https://github.com/Sloeber/arduino-eclipse-plugin/assets/1175841/dd1d8188-98b6-4922-901b-5029b3b5276a)

The default sketch builds on my system
![image](https://github.com/Sloeber/arduino-eclipse-plugin/assets/1175841/2541beba-5e47-47ae-bd81-684cb019a05a)

**Comment by tpiuser** [comment]
Thanks, I did this but the install err'ed-out and the first compile of the default code couldn't find the c++ compiler.  
I looked in the download folder under the Sloeber/arduinoPlugin folder and found the compiler tarball.  
I extracted it where the ide was looking for it and it compiled as expected.  I was hoping to find out if anything else may be missing.
(libraries to link against)

**Comment by tpiuser** [comment]
It looks like also the default includes are missing under C/C++ General/Preprocessor Include Paths.
All it says is Arduino Compiler Settings but nothing under it and it won't allow for 'Add' or 'Edit', all are greyed out.

**Reply by jantje** [reply]
I would remove the renesas_uno folder and reinstall using the boards manager

**Comment by tpiuser** [comment]
I had to add a 'User Defined Toolchains' in Window/Preferences/C/C++/Core Build Toolchains  that points to the tarball I extracted earlier.
Then the Arduino Compiler Settings was populated.

**Reply by jantje** [reply]
That is definitely not the way to go.
Please try something easier so you know how to do things.
>Window/Preferences/C/C++/Core Build Toolchains
is not part of Sloeber and should not impact sloeber in any way

**Reply by tpiuser** [reply]
The anomaly  came from: 
1.  Installing the new board and hitting apply and close before it indicated it was complete on the taskbar.
2. And, not restarting eclipse.

I tend to expect eclipse to throw up a "Restart now" message.  It's probably already in a read-me file somewhere.

If I had followed the steps you laid out, waited for the process to complete then restarted, I may not have seen the issue at all.

Great plugin.  
Thanks.
